File: Benchmarking/src/gen390.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s7 {
    var p0: Double
    var p1: s0
  }
  @inline(never)
  func f0(_ v0: s7, _ v1: Double) -> Double {
    let v6: Double = v1 + v1
    let v3: [Double] = [v1, v6, v6]
    let v5: Double = v0.p0
    var v10: [Double] = v3
    let v7: Double = v10[2]
    let v28: Double = v3[2]
    v10[1] = v28
    v10[2] = v1
    let v31: Double = v10[2]
    let v21: Double = v28 * v31
    let v25: [Double] = [v7, v5, v5, v5]
    let v53: Double = v25[0]
    v10[2] = v21
    return v53
  }
  func benchmark() {
    let v0: s7 = s7(p0: 0.0, p1: s0(p0: [[1.0], [2.0], [3.0]], p1: [[4.0], [5.0]]))
    let v1: Double = 6.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 6.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen390.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s7 (
    p0: Double,
    p1: s0
  )
  @noinline
  def f0(v0: s7, v1: Double): Double = {
    val v6: Double = v1 + v1
    val v3: Vector[Double] = Vector(v1, v6, v6)
    val v5: Double = v0.p0
    var v10: Vector[Double] = v3
    val v7: Double = v10(2)
    val v28: Double = v3(2)
    v10 = v10.updated(1, v28)
    v10 = v10.updated(2, v1)
    val v31: Double = v10(2)
    val v21: Double = v28 * v31
    val v25: Vector[Double] = Vector(v7, v5, v5, v5)
    val v53: Double = v25(0)
    v10 = v10.updated(2, v21)
    v53
  }
  def benchmark(): Unit = {
    val v0: s7 = s7(0.0, s0(Vector(Vector(1.0), Vector(2.0), Vector(3.0)), Vector(Vector(4.0), Vector(5.0))))
    val v1: Double = 6.0
    val start = nanoTime()
    var result: Double = 6.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}